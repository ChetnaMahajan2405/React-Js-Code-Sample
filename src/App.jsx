// This component handles the App template used on every page.
import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import ReduxToastr from "react-redux-toastr";

import appRoutes from './config/routes';

const App = ({ authenticated, checked }) => {
  return (
    <div>
      <ReduxToastr
        timeOut={4000}
        transitionIn="fadeIn"
        transitionOut="fadeOut" />
      {appRoutes(authenticated, checked)}
    </div>
  );
}

const { bool } = PropTypes;

App.propTypes = {
  authenticated: bool.isRequired,
  checked: bool.isRequired
};

function mapStateToProps({ session }) {
  return {
    checked: session.checked,
    authenticated: session.authenticated,
  }
}

export default connect(mapStateToProps, null)(App);
