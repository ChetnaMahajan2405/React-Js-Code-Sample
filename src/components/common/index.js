
export { default as Button } from "./Button";
export { default as CheckBox } from "./CheckBox";
export { default as Header } from "./Header";
export { default as Image } from "./Image";
export { default as TextInput } from "./TextInput";
export { default as Tile } from "./Tile";