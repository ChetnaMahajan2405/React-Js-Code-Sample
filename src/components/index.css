
.form-box {
    margin: auto;
    background-color: #ffffff;
    border: 3px solid #f6f6f6;
    padding-top: 16px; 
    padding-bottom: 25px;
};

@media all and (min-width : 1200px) {
    .form-box {
        width: 400px;
        padding-top: 31px; 
        padding-bottom: 40px;
    }
}

@media all and (min-width : 768px) {
    .form-box {
        width: 400px;
        padding-top: 31px; 
        padding-bottom: 40px;
    }
}

@media all and (min-width: 600px) {
    .form-box {
        width: 400px;
        padding-top: 31px; 
        padding-bottom: 40px;
    }

}  

div .form-box .form-header {
    font-size: 24px;
    color: #414141;
    font-family: "Open Sans";
    font-weight: 600;
    text-align: center;
}

div .form-box .form-subHeading {
    font-size: 14px;
    color: #333333;
    font-family: "Open Sans";
    font-weight: 400;
    text-align: center;
}

div .form-box .input-container { 
    padding-left: 20px;
    padding-right: 20px;
    margin-top: 35px;
}

@media all and (max-width: 600px) {
    div .form-box .input-container  {
        padding-left: 10px;
        padding-right: 10px;
    }

}

.carousel {
    margin-top: 141px;
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

@media all and (max-width: 600px) {
    .carousel {
        margin-top: 100px;
    }
}


/* Create two equal columns that floats next to each other */
.column {
    float: left;
    width: 50%;
}