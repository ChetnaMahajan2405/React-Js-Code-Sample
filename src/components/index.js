export { default as Forgot } from "./Forgot";
export { default as Home } from "./Home";
export { default as Login } from "./Login";
