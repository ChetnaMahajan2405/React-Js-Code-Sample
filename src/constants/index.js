export {
  baseUrl,
  __DEV__,
  data, 
} from "./appConstants";