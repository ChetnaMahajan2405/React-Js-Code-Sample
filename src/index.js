
import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import { MainLayout } from "./layouts";
import App from "./App";
import store from "./redux/store/configStore";


const wrapApp = reduxStore => (
  <MainLayout>
    <Provider store={reduxStore}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </Provider>
  </MainLayout>
);

const rootEl = document.getElementById("root");

ReactDOM.render(wrapApp(store), rootEl);
