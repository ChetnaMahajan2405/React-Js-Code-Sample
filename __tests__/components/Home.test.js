import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

// Component to be tested
import { Home } from '../../src/components';
import { Button } from '../../src/components/common';

const initialState = {
  session: {
    autheticated: true,
    checked: true,
    user: {
      name: 'Luke Skywalker'
    }
  },
  search: {
    data: []
  }
};

const mockedStore = configureStore(initialState);

jest.mock('../../src/redux/services/service');

describe('<Home /> renders()', () => {
  it('Home component spanshot', () => {
    const wrapper = shallow(
      <Home store={mockedStore} />
    );
    const component = wrapper.dive();
    expect(toJson(component)).toMatchSnapshot();
  });


});
