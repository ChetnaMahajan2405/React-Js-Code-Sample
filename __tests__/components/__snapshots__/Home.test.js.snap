// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders() Home component spanshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
